prow/github/client.go

```diff
@@ -386,6 +386,24 @@ func (c *Client) CreateStatus(org, repo, ref string, s Status) error {
 	return err
 }
 
+func (c *Client) GetLabels(org, repo string) ([]string, error) {
```

Comment:
When you call AddLabels it will return a list of labels on the issue. If that list doesn't contain the labels you tried to add then they don't exist on the repo. You can use that instead of a call to GetLabels.